src/kOS/Screen/TermWindow.cs

```diff
@@ -549,11 +549,20 @@ public int GetFontSize()
         /// </summary>
         private void ProcessTelnetInput()
         {
-            foreach (var telnet in telnets)
+            // It's possible to close and remove telnets from the list during the processing
+            // of input (if the detach signal Ctrl-D is sent).  Therefore we have to
+            // make this temp copy to prorect against the C# error "Collection was Modified"
+            // during the foreach loop:
```

Comment:
You'll see this comment in several places in this commit.  Anywhere that something fitting the following pattern was used:
```
foreach (var telnet in telnets)
{
   do something that could result in detaching the telnet from the telnets list.
}
```
The telnet list can have one of the telnets removed from it during the right kinds of keypresses, or when stopping the listener, so it's not safe to ``foreach`` through it while that can happen.  This problem was probably always there but not noticed because I wasn't running the debug EXE of Kerbal, so I'd only notice if I looked at the output log. (where the debug version shows the exception right away on screen as it happens so it's blatant).  The reason I never saw any major failures before was that this was always happening just as things were detaching and going away so most of the bad state of data caused by the exception aborting the work partway through didn't matter.  (I was constructing a new instance anyway, not re-using the same one like I am now.)